utils/heatmap_dataset.py

```diff
@@ -73,18 +76,31 @@ def _sample_bounding_box(self, shape):
             for sh, psh in zip(shape, self.patch_shape)
         ]
         return tuple(slice(start, start + psh) for start, psh in zip(bb_start, self.patch_shape))
-        
+
     def _get_sample(self, index):
         if self.sample_random_index:
             index = np.random.randint(0, len(self.raw_images))
         raw, label = self.raw_images[index], self.label_images[index]
 
-        #TODO this is specific for challenge zarr files now, maybe need to generalize in the future
-        #zarr_file = zarr.open(f"{raw}/VoxelSpacing10.000/denoised.zarr/0", mode='r')
+        # TODO this is specific for challenge zarr files now, maybe need to generalize in the future
+        # Yes ;). We can discuss this soon.
+
+        # zarr_file = zarr.open(f"{raw}/VoxelSpacing10.000/denoised.zarr/0", mode='r')
         zarr_file = zarr.open(os.path.join(raw, "VoxelSpacing10.000", "denoised.zarr", "0"), mode='r')
-        raw = zarr_file[:]
-        #sigma is not really used in my process_tomogram ... TODO ?
-        label = process_tomogram(label, raw.shape, eps=self.eps, sigma=self.sigma, lower_bound=self.lower_bound, upper_bound=self.upper_bound)
+
+        # This was very inefficient!
+        # You first load the full data from zarr and then later load the bounding box.
+        # raw = zarr_file[:]
+        # Instead, you can just load the bounding box from the zarr
+        raw = zarr_file
```

Comment:
Your previous code was quite inefficient because it always loaded the full data from the zarr file, although we just need to load the bounding box.